openhtf/io/frontend/src/app/utils.ts

```diff
@@ -0,0 +1,206 @@
+// Copyright 2016 Google Inc.All Rights Reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+import {Pipe, PipeTransform} from 'angular2/core';
+
+
+// TODO(jethier): Clean up all these maps, maybe using inheritance. They are
+//                all doing the same type of operation, just on different data.
+
+/** A Pipe for turning status codes into materialize-css color classes.**/
+@Pipe({name: 'statusToColor'})
+export class StatusToColor implements PipeTransform {
+  statusMap = {
+    'RUNNING': 'blue lighten-1 white-text',
+    'EXECUTING': 'blue lighten-1 white-text',
+    'ONLINE':  'blue lighten-1 white-text',
+    'PASS':    'green lighten-1 white-text',
+    'FAIL':    'red lighten-1 white-text',
+    'ERROR':   'yellow darken-3 white-text',
+    'PENDING': 'blue-grey lighten-3 white-text',
+    'START_WAIT': 'blue-grey lighten-3 white-text',
+    'STOP_WAIT': 'blue-grey lighten-3 white-text',
+    'OFFLINE': 'blue-grey lighten-3 white-text',
+  };
+
+  /**
+   * Return a materialize-css color class string for the given status code.
+   * @param {string} status - A string representing a status code.
+   * @return {string} A string of materialize-css color classes.
+   */
+  transform(status: string) : string {
+    return this.statusMap[status]; 
+  }
+}
+
+
+/** A Pipe for darkening status colors. **/
+@Pipe({name: 'colorToDark'})
+export class ColorToDark implements PipeTransform {
+  darkStatusMap = {
+    'blue lighten-1 white-text':      'blue darken-2 white-text',
+    'green lighten-1 white-text':     'green darken-2 white-text',
+    'red lighten-1 white-text':       'red darken-2 white-text',
+    'yellow darken-3 white-text':     'yellow darken-4 white-text',
+    'blue-grey lighten-3 white-text': 'blue-grey lighten-2 white-text'
+  };
+
+  /**
+   * Return darker versions of the given status color classes.
+   * @param {string} color - materialize-css color classes for a status.
+   * @return {string} Darker versions of the given color classes.
+   */
+  transform(color: string) : string {
+    return this.darkStatusMap[color];
+  }
+}
+
+
+/** A Pipe for darkening status colors even more. **/
```

Comment:
On one hand I was sort of hoping to come up with something more robust that would also allow me to define the whole color palette in one spot (right now some colors are pulled from here, some are in html, and some are hard-coded in css). On the other hand, the MapPipe base class does clean this up quite a bit, so I think I'm just gonna go with the base class concept for now. Thanks!